### ISSUE TITLE
Error while running build script

### ISSUE THREAD

**Issue by clement911** [dotnet/efcore#33155]
I'm trying to get EF core to build and run locally. 

I found the following page: https://github.com/dotnet/efcore/blob/main/docs/getting-and-building-the-code.md

One of the first step is to run the `build` command. However, I get some errors:

```
C:\Dev\efcore>build.cmd
  Determining projects to restore...
  Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/a65e5cb4-26c0-410f-9457-06d
  b3c5254be/nuget/v3/flat2/microsoft.signed.wix/index.json'.
  The SSL connection could not be established, see inner exception.
    Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host..
    An existing connection was forcibly closed by the remote host.
  Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/a65e5cb4-26c0-410f-9457-06d
  b3c5254be/nuget/v3/flat2/vswhere/index.json'.
  The SSL connection could not be established, see inner exception.
    Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host..
    An existing connection was forcibly closed by the remote host.
  Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242
  650c914a8/nuget/v3/flat2/vswhere/index.json'.

.........

  Failed to download package 'MicroBuild.Core.0.2.0' from 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-4
  7c8-91e5-8ec20c22b4f8/nuget/v3/flat2/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg'.
  The SSL connection could not be established, see inner exception.
    Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..
    An existing connection was forcibly closed by the remote host.
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : Failed to download package 'MicroBuild.Core.Sentinel.1.0.0' from 'https://pkgs.dev.azure.com
/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/microbuild.core.sentinel/1.0.0/microbuild.core.s
entinel.1.0.0.nupkg'. [C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : The SSL connection could not be established, see inner exception. [C:\Users\cleme\.nuget\pac
kages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   Unable to read data from the transport connection: An existing connection was forcibly clo
sed by the remote host.. [C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   An existing connection was forcibly closed by the remote host. [C:\Users\cleme\.nuget\pack
ages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : Failed to download package 'MicroBuild.Core.0.2.0' from 'https://pkgs.dev.azure.com/dnceng/9
ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg'. [C:\U
sers\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : The SSL connection could not be established, see inner exception. [C:\Users\cleme\.nuget\pac
kages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   Unable to read data from the transport connection: An existing connection was forcibly clo
sed by the remote host.. [C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   An existing connection was forcibly closed by the remote host. [C:\Users\cleme\.nuget\pack
ages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
  Failed to restore C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj (in 13.23 sec).

Build FAILED.

C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : Failed to download package 'MicroBuild.Core.Sentinel.1.0.0' from 'https://pkgs.dev.azure.com
/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/microbuild.core.sentinel/1.0.0/microbuild.core.s
entinel.1.0.0.nupkg'. [C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : The SSL connection could not be established, see inner exception. [C:\Users\cleme\.nuget\pac
kages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   Unable to read data from the transport connection: An existing connection was forcibly clo
sed by the remote host.. [C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   An existing connection was forcibly closed by the remote host. [C:\Users\cleme\.nuget\pack
ages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : Failed to download package 'MicroBuild.Core.0.2.0' from 'https://pkgs.dev.azure.com/dnceng/9
ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg'. [C:\U
sers\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error : The SSL connection could not be established, see inner exception. [C:\Users\cleme\.nuget\pac
kages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   Unable to read data from the transport connection: An existing connection was forcibly clo
sed by the remote host.. [C:\Users\cleme\.nuget\packages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
C:\Dev\efcore\.dotnet\sdk\8.0.101\NuGet.targets(156,5): error :   An existing connection was forcibly closed by the remote host. [C:\Users\cleme\.nuget\pack
ages\microsoft.dotnet.arcade.sdk\9.0.0-beta.23607.2\tools\Tools.proj]
    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:13.44
Build failed with exit code 1. Check errors above.
```

**Comment by clement911** [comment]
This is what I see from Chrome:

![image](https://github.com/dotnet/efcore/assets/3426504/94457de1-71c3-471f-bd9b-d3e7c2a0f682)

**Comment by roji** [comment]
@clement911 that looks like a network-related issue, that could be a momentary/transient issue; it does not reproduce here. In any case, it isn't related to EF itself, but is rather a nuget error.

**Comment by clement911** [comment]
That's weird. I tried again today and was still getting loads of errors.
(Note that I never have no issues when using nuget via VS)

As I was investigating the issue with Fiddler, I notice a ton of HTTP errors, mostly 404 and 303.

I ran build.cmd numerous times to investigate. Each time, the script manage to download a few more packages.
Eventually after running it like 10 times, it had all the packages downloaded in cache and the build completed. Yay! 🙂

I'm not sure why this happened but thanks anyway @roji 
Closing.
